packages/contracts-bedrock/deploy/018-MigrationSystemDictatorSteps.ts

```diff
@@ -78,88 +113,251 @@ const deployFn: DeployFunction = async (hre) => {
     await awaitCondition(async () => {
-      const owner = await L1StandardBridge.callStatic.getOwner(getOwnerOpts)
+      const owner = await L1StandardBridgeProxy.callStatic.getOwner({
+        from: ethers.constants.AddressZero,
+      })
       return owner === MigrationSystemDictator.address
     })
   } else {
-    console.log(`L1StandardBridge already owned by the MigrationSystemDictator`)
+    console.log(`L1StandardBridge already owned by MSD`)
+  }
+
+  const checks = {
+    1: async () => {
+      await assertContractVariable(
+        ProxyAdmin,
+        'addressManager',
+        AddressManager.address
+      )
+      assert(
+        (await ProxyAdmin.implementationName(
+          getDeploymentAddress(hre, 'Proxy__OVM_L1CrossDomainMessenger')
+        )) === 'OVM_L1CrossDomainMessenger'
+      )
+      assert(
+        (await ProxyAdmin.proxyType(
+          getDeploymentAddress(hre, 'Proxy__OVM_L1CrossDomainMessenger')
+        )) === 2
+      )
+      assert(
+        (await ProxyAdmin.proxyType(
+          getDeploymentAddress(hre, 'Proxy__OVM_L1StandardBridge')
+        )) === 1
+      )
+    },
+    2: async () => {
+      await assertContractVariable(L1CrossDomainMessenger, 'paused', true)
+      const deads = [
+        'Proxy__OVM_L1CrossDomainMessenger',
+        'Proxy__OVM_L1StandardBridge',
+        'OVM_CanonicalTransactionChain',
+        'OVM_L2CrossDomainMessenger',
+        'OVM_DecompressionPrecompileAddress',
+        'OVM_Sequencer',
+        'OVM_Proposer',
+        'OVM_ChainStorageContainer-CTC-batches',
+        'OVM_ChainStorageContainer-CTC-queue',
+        'OVM_CanonicalTransactionChain',
+        'OVM_StateCommitmentChain',
+        'OVM_BondManager',
+        'OVM_ExecutionManager',
+        'OVM_FraudVerifier',
+        'OVM_StateManagerFactory',
+        'OVM_StateTransitionerFactory',
+        'OVM_SafetyChecker',
+        'OVM_L1MultiMessageRelayer',
+      ]
+      for (const dead of deads) {
+        assert(
+          (await AddressManager.getAddress(dead)) ===
+            ethers.constants.AddressZero
+        )
+      }
+    },
+    3: async () => {
+      await assertContractVariable(AddressManager, 'owner', ProxyAdmin.address)
+      assert(
+        (await L1StandardBridgeProxy.callStatic.getOwner({
+          from: ethers.constants.AddressZero,
+        })) === ProxyAdmin.address
+      )
+    },
+    4: async () => {
+      // Check L2OutputOracle was initialized properly.
+      await assertContractVariable(
+        L2OutputOracle,
+        'latestBlockNumber',
+        hre.deployConfig.l2OutputOracleStartingBlockNumber
+      )
+      await assertContractVariable(
+        L2OutputOracle,
+        'proposer',
+        hre.deployConfig.l2OutputOracleProposer
+      )
+      await assertContractVariable(
+        L2OutputOracle,
+        'owner',
+        hre.deployConfig.l2OutputOracleOwner
+      )
+      if (
+        hre.deployConfig.l2OutputOracleGenesisL2Output !==
```

Comment:
Hmm on the prod network it should not be `bytes32(0)`

Reply:
Added a big warning for this